core/services/gateway/handlers/vault/handler.go

```diff
@@ -269,12 +271,22 @@ func (h *handler) HandleNodeMessage(ctx context.Context, resp *jsonrpc.Response[
 	}
 
 	ar.mu.Lock()
-	ar.responses = append(ar.responses, resp)
+	_, exists := ar.responses[nodeAddr]
+	if exists {
+		l.Errorw("duplicate response from node, ignoring", "nodeAddr", nodeAddr)
+		ar.mu.Unlock()
+		return nil
+	}
+	ar.responses[nodeAddr] = resp
 	ar.mu.Unlock()
 
 	resp, err := h.aggregator.Aggregate(ctx, l, ar, resp)
-	if err != nil {
-		l.Warnw("error aggregating responses", "error", err)
+	switch {
+	case errors.Is(err, errQuorumUnobtainable):
+		l.Error("quorum unobtainable, returning response to user...", "error", err)
+		return h.sendResponse(ctx, ar, h.errorResponse(ar.req, api.FatalError, err))
+	case err != nil:
+		l.Warnw("error aggregating responses, waiting for other nodes...", "error", err)
```

Comment:
This should be debug or info log, coz it's expected to be logged for each request multiple times before reaching quorum.